### ISSUE TITLE
train_*_estimator flags in hybrid mutual information estimator

### ISSUE THREAD

**Issue by bkmi** [mfederici/torch-mist#5]
As you can see here, there are flags for training different parts of the hybrid mutual information estimator.

https://github.com/mfederici/torch-mist/blob/14c130f458a45847ff21dd33f1bb2e3db9282464/src/torch_mist/estimators/hybrid.py#L16-L17

I wanted to ask, why not control this via `hybrid_estimator.generative_estimator.parameters()` or `hybrid_estimator.discriminative_estimator.parameters()` and passing them to the correct optimizer. Then the actual value of the loss would be consistent with, say, previous iterations where one of the estimators was not trained. Is it for efficiency?

imo having multiple methods for doing this may add confusion.

**Comment by mfederici** [comment]
The main reason is efficiency. If we know we don't need to train one or the other component we can simplify a lot of computation

**Comment by mfederici** [comment]
This has been implemented now. The `torch_mist.utils.freeze` submodule provides utilities to freeze a `nn.Module` and check if they are trainable by setting and checking `requires_grad`
